src/test/java/org/kie/processmigration/service/MigrationServiceImplTest.java

```diff
@@ -16,675 +16,239 @@
-        } finally {
-            getEntityManager().getTransaction().commit();
-        }
-    }
-
-    @Test
-    public void testSubmitPlanNotFound() throws InvalidMigrationException {
-        // Given
-        Plan plan = createPlan();
-        MigrationDefinition def = createMigrationDefinition(plan, SYNC);
-        def.setPlanId(9999L);
-
-        // Setup mock
-        ProcessAdminServicesClient mockProcessAdminServicesClient = Mockito.mock(ProcessAdminServicesClient.class);
-        Mockito.when(kieServiceMock.getProcessAdminServicesClient(anyString())).thenReturn(mockProcessAdminServicesClient);
-        addMockConfigs(kieServiceMock);
-
-        // When
-        try {
-            getEntityManager().getTransaction().begin();
-            migrationService.submit(def);
-            fail("Expected InvalidMigrationException due to wrong planId");
-        } catch (InvalidMigrationException e) {
-            assertNotNull(e);
-        } finally {
-            getEntityManager().getTransaction().commit();
-        }
-    }
-
-    @Test(expected = InvalidMigrationException.class)
-    public void testSubmitValidationNullDefinition() throws InvalidMigrationException {
-        migrationService.submit(null);
-    }
+        List<Migration> migrations = migrationService.findAll();
 
-    @Test(expected = InvalidMigrationException.class)
-    public void testSubmitValidationNullPlan() throws InvalidMigrationException {
-        migrationService.submit(new MigrationDefinition());
+        assertThat(migrations, notNullValue());
+        assertThat(migrations, hasSize(1));
+        Migration migration = migrations.get(0);
+        assertThat(migration.getStatus(), is(Execution.ExecutionStatus.COMPLETED));
+        assertThat(migration.getCancelledAt(), nullValue());
+        assertThat(migration.getStartedAt(), notNullValue());
+        assertThat(migration.getFinishedAt(), notNullValue());
+        List<MigrationReport> results = migrationService.getResults(migration.id);
+        assertThat(results, hasSize(1));
+        assertThat(results.get(0).getProcessInstanceId(), is(report.getProcessInstanceId()));
+        assertThat(results.get(0).getSuccessful(), is(report.isSuccessful()));
+        assertThat(results.get(0).getStartDate(), is(report.getStartDate().toInstant()));
+        assertThat(results.get(0).getEndDate(), is(report.getEndDate().toInstant()));
+        assertThat(results.get(0).getMigrationId(), is(migration.id));
+        assertThat(results.get(0).getLogs(), containsInAnyOrder(report.getLogs().toArray()));
+
+        verify(mockAdminServicesClient, times(1)).migrateProcessInstance(anyString(), anyLong(), anyString(), anyString(), anyMap());
+        migrationService.delete(migration.id);
     }
 
     @Test
-    public void testSubmitValidationInvalidKieServer() throws InvalidMigrationException {
+    void testSubmitSyncFailedMigration() throws InvalidMigrationException, PlanNotFoundException, MigrationNotFoundException {
```

Comment:
Maybe you could reuse the previous test and just call it passing a report result parameter set to `false`, so we can avoid duplicating code here.